SongSnap_ImageUnfrozen (1).ipynb

```diff
@@ -0,0 +1,4349 @@
+        "        total_precision_pop += precision_score(preds_pop.cpu(), batch_y_popularity.cpu(), average = 'macro')\n",
+        "        total_recall_pop += recall_score(preds_pop.cpu(), batch_y_popularity.cpu(), average = 'macro')\n",
+        "        total_f1_pop += f1_score(preds_pop.cpu(), batch_y_popularity.cpu(), average = 'macro')\n",
+        "\n",
+        "        # Save the outputs to lists\n",
+        "        all_genre_outputs.extend(preds_genre.cpu().numpy())\n",
+        "        all_popularity_outputs.extend(preds_pop.cpu().numpy())\n",
+        "        all_danceability_outputs.extend(danceability_output.cpu().numpy())\n",
+        "        all_energy_outputs.extend(energy_output.cpu().numpy())\n",
+        "\n",
+        "        num_batches += 1\n",
+        "\n",
+        "    # Calculate average loss\n",
+        "    avg_loss_genre = total_loss_genre / num_batches\n",
+        "    avg_loss_popularity = total_loss_popularity / num_batches\n",
+        "    avg_loss_danceability = total_loss_danceability / num_batches\n",
+        "    avg_loss_energy = total_loss_energy / num_batches\n",
+        "\n",
+        "    avg_acc_genre = total_accuracy_genre / num_batches\n",
+        "    avg_precision_genre = total_precision_genre / num_batches\n",
+        "    avg_recall_genre = total_recall_genre / num_batches\n",
+        "    avg_f1_genre = total_f1_genre / num_batches\n",
+        "\n",
+        "    avg_acc_pop = total_accuracy_pop / num_batches\n",
+        "    avg_precision_pop = total_precision_pop / num_batches\n",
+        "    avg_recall_pop = total_recall_pop / num_batches\n",
+        "    avg_f1_pop = total_f1_pop / num_batches\n",
+        "\n",
+        "    # Calculate the average accuracy, recall, precision and F1 score for genre & popularity\n",
+        "\n",
+        "\n",
+        "    print(\"Evaluation Results:\")\n",
+        "    print(\"Average Genre loss: {:.4f}\".format(avg_loss_genre))\n",
+        "    print(\"Average Popularity loss: {:.4f}\".format(avg_loss_popularity))\n",
+        "    print(\"Average Danceability loss: {:.4f}\".format(avg_loss_danceability))\n",
+        "    print(\"Average Energy loss: {:.4f}\".format(avg_loss_energy))\n",
+        "\n",
+        "    print(\" \")\n",
+        "    print(\"Average Accuracy Popularity: {:.4f}\".format(avg_acc_pop))\n",
+        "    print(\"Average Precision Popularity: {:.4f}\".format(avg_precision_pop))\n",
+        "    print(\"Average Recall Popularity: {:.4f}\".format(avg_recall_pop))\n",
+        "    print(\"Average F1 Popularity: {:.4f}\".format(avg_f1_pop))\n",
+        "\n",
+        "    print(\" \")\n",
+        "    print(\"Average Accuracy Genre: {:.4f}\".format(avg_acc_genre))\n",
+        "    print(\"Average Precision Genre: {:.4f}\".format(avg_precision_genre))\n",
+        "    print(\"Average Recall Genre: {:.4f}\".format(
```

Comment:
the output format for danceability and energy is a string that looks like a list. would it be possibly to turn it into a float?

Reply:
hmm i think it is a float when it is calculated?